tests/comments/flow_union.js

```diff
@@ -0,0 +1,9 @@
+type UploadState<E, EM, D>
+  // The upload hasnt begun yet
```

Comment:
I feel like it's better that we encourage this:

```js
type UploadState<E, EM, D> =
  // comment
  | { type: "etc" }
  // comment
  | { type: "etc etc"}
```

That seems a lot more consistent and would make the code in this PR simpler, right?